### PR TITLE
Configure Jest and Cypress with React testing library

### PR DESCRIPTION
**Pull request by vassalloandrea**
Quick Info
---

| Issue |
| -- | 
| #7  |

- Install and configure Jest with React testing library
- Install and configure Cypress
- Run Cypress using a test server which is different from the development one
- Disable the Apollo client cache when the server is run with the `test` ENV variable
- Create the Cypress action to allow it to stub GraphQL requests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,27 @@
   npm run dev
 ```
 
+### Running the unit test
```
Comment (dinohamzic) [inline README.md:13]: We should change this to:
> Running unit tests

File: README.md [hunk 2]
```diff
@@ -10,6 +10,27 @@
...
+
+Solidus NextJS Frontend uses Jest to run unit test with the help
```
Comment (dinohamzic) [inline README.md:15]: > ...to run unit test**s**...

File: README.md [hunk 3]
```diff
@@ -10,6 +10,27 @@
...
+of React testing library to render the components.
+
+You can run the unit tests locally using:
```
Comment (dinohamzic) [inline README.md:18]: > You can run ~~the~~ unit tests locally using:

File: README.md [hunk 4]
```diff
@@ -10,6 +10,27 @@
...
+
+```
+npm run test:unit
+```
+
+### Running the E2E test
+
+Solidus NextJS Frontend uses Cypress to run the E2E test.
```
Comment (dinohamzic) [inline README.md:26]: > ...to run ~~the~~ E2E test**s**...

File: README.md [hunk 5]
```diff
@@ -10,6 +10,27 @@
...
+Cypress usually is run locally using the development server but in this case, we use the `start server and run` library to run a dedicated `test` server which is used by Cypress.
```
Comment (dinohamzic) [inline README.md:27]: > Cypress is usually run locally using the development server. However, in this case, we use the `start server and run` library to run a dedicated `test` server for Cypress.

File: README.md [hunk 6]
```diff
@@ -10,6 +10,27 @@
...
+
+This is needed to disable Apollo client cache, in this way we can
```
Comment (dinohamzic) [inline README.md:29]: > This is needed to disable Apollo's client cache and allows us to stub GraphQL requests.

**Review by dinohamzic — APPROVED** [review]

**Review by AlessioRocco — APPROVED** [review]
🚀

**Pull request merged**